ykrt/src/compile/jitc_yk/jit_ir.rs

```diff
@@ -0,0 +1,400 @@
+//! The Yk JIT IR
+
+use std::fmt;
+use strum_macros::FromRepr;
+
+/// Number of bits used to encode an opcode.
+const OPCODE_SIZE: u64 = 8;
+
+/// Max number of operands in a short instruction.
+const SHORT_INSTR_MAX_OPERANDS: u64 = 3;
+
+/// Bit fiddling for a short operands.
+const SHORT_OPERAND_SIZE: u64 = 18;
+const SHORT_OPERAND_KIND_SIZE: u64 = 3;
+const SHORT_OPERAND_KIND_MASK: u64 = 7;
+const SHORT_OPERAND_MASK: u64 = 0x3ffff;
+
+/// Bit fiddling for instructions.
+const INSTR_ISSHORT_SIZE: u64 = 1;
+const INSTR_ISSHORT_MASK: u64 = 1;
+const INSTR_OPCODE_MASK: u64 = 0xe;
+
+/// An instruction is identified by its index in the instruction vector.
+pub(crate) type InstructionID = usize;
```

Comment:
In general I've learnt not to use `type` like this, because it gives the illusion that `usize` and `InstructionID` can't be confused --- but Rust considers them exactly the same type. If we want to do this, a new type (e.g. `struct InstructionId(usize)`) actually gives you a degree of type safety.

Reply:
Fixed in 6cd71d2cc456d260ee0f69a64c6ca69f3ae2ae5a